blogposts/2022/release-3.39

```diff
@@ -0,0 +1,75 @@
+    url: https://github.com/sourcegraph/sourcegraph/pull/33308
+    category: Search
+  - description: "Built-in authentication (i.e. username and password) now supports account lockout after consecutive failed sign-in attempts. New config options have been added under `\auth.lockout\` to customize the threshold for failed attempts and length of lockouts."
+    url: https://github.com/sourcegraph/sourcegraph/pull/33999
+    category: Admin
+---
+
+Sourcegraph 3.39 is now available! In this release:
+
+## Onboard developers fast with Notebooks, now Generally Available
+
+Built with onboarding in mind, Notebooks allow you to integrate search results and Markdown to make documentation that is both effortless and doesn't get stale. Notebooks are inspired by the Jupyter Notebooks project and feature four unique block types, an intelligent notepad for creating notebooks on the fly, and deep integration with Code Search so you can always find the code you need.
+
+<video class="blog-image" title="Notebooks overview" alt="An example of using a notebook." loop autoplay muted playsinline>
+  <source src="https://user-images.githubusercontent.com/1657213/163855878-4355bcb0-5407-4307-b3f2-4581e1ab2fe2.mp4">
+</video>
+
+Web-based notebooks are the easiest way to create and share notebooks with your team or company, but we also support file-based notebooks via a special `.snb.md` file type. With file-based Notebooks, you can transform your existing documentation and augment it with Sourcegraph query blocks. You can also embed notebooks on any web page, enabling easy integration with your own knowledge management systems, so you can meet your developers where they are.
+
+Notebooks are now available for all users in Sourcegraph 3.39, and you can also explore some of our [public notebooks on Sourcegraph Cloud](https://sourcegraph.com/notebooks?tab=explore) with no account required.
+
+To learn more about Notebooks, [read the docs](https://docs.sourcegraph.com/notebooks).
+
+## Initial support for filtering code insights using search contexts
+
+<div style="text-align:center">
+<img class="blog-image" title="Code insights have search context filters" alt="Image of a search context being used to filter a code insight" src="https://storage.googleapis.com/sourcegraph-assets/blog/3.39/code_insights_3.39_dynamic_context_filters.png
+">
+</div>
+
+Using a search context lets you reuse the same `repo:` pattern to filter multiple insights to a subset of your repos, and then maintain and update that group of repos in just one place, the search context. Your filtered insights will automatically reflect the current state of any updated context. Using a context is also a more succinct way to reference an extended `repo:` filter for reuse as a filter.
```

Comment:
```suggestion
Using a search context lets you reuse the same `repo:` pattern to filter multiple insights to a subset of your repos, and then maintain and update that group of repos in just one place, the search context. Your filtered insights will automatically reflect the current state of any updated context. Using a context is also a more succinct way to reference an extended `repo:` filter for reuse as a filter.

```

Reply:
Yes — I think we can do both of those things. @abschu @Joelkw could you let me know if this reads more easily?

```suggestion
[Search contexts](https://docs.sourcegraph.com/code_search/explanations/features#search-contexts) are pre-set repository groups that make it easy to scope your Sourcegraph searches, and you can now apply search contexts to Code Insights as well. You can use a search context to apply the `repo:` filter from that context to the data returned in your insights. Then, by updating your search contexts, your filtered insights will automatically update to reflect the changes as well. Using search contexts is a fast, succinct way to reuse complex repository filters.

```

Reply:
I removed the mention of `repo:` fields because that's covered in the next paragraph, otherwise like it!